src/internal/components/dropdown/__integ__/alignment.test.ts

```diff
@@ -6,28 +6,47 @@ import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
+  describe.each(['expandable', 'expandable-iframe'])('%s', pageName => {
+    describe.each([true, false])('expandToViewport=%s', expandToViewport => {
+      const alignments = ['top-left', 'top-right', 'bottom-left', 'bottom-right'] as const;
 
-    function setupTest(alignment: Alignment, testFn: (page: BasePageObject) => Promise<void>) {
-      return useBrowser(async browser => {
-        await browser.url(`#/light/dropdown/expandable?expandToViewport=${expandToViewport}`);
-        const page = new BasePageObject(browser);
-        await page.waitForVisible(createWrapper().findAutosuggest().toSelector());
-        await testFn(page);
-      });
-    }
+      function setupTest(testFn: (page: BasePageObject) => Promise<void>) {
+        return useBrowser(async browser => {
+          const runInIframe = async (callback: () => Promise<void>) => {
+            if (pageName !== 'expandable-iframe') {
+              // noop if the page does not use iframes
+              return callback();
+            }
+            const iframeEl = await browser.$('#expandable-dropdowns-iframe');
+            await iframeEl.waitForDisplayed();
+            await browser.switchToFrame(iframeEl);
+            await callback();
+            // go back to top
+            await browser.switchToFrame(null);
+          };
+          await browser.url(`#/light/dropdown/${pageName}?expandToViewport=${expandToViewport}`);
+          const page = new BasePageObject(browser);
+          await runInIframe(async () => {
+            await page.waitForVisible(createWrapper().findAutosuggest().toSelector());
+            await testFn(page);
+          });
+        });
+      }
 
-    test.each(alignments)('alignment at %s corner', alignment => {
-      return setupTest(alignment, async page => {
-        const wrapper = createWrapper().findAutosuggest(`#${alignment}`);
-        await page.click(wrapper.findNativeInput().toSelector());
-        const dropdownBox = await page.getBoundingBox(wrapper.findDropdown({ expandToViewport }).toSelector());
-        const triggerBox = await page.getBoundingBox(wrapper.findNativeInput().toSelector());
-        expect(dropdownBox.left).toEqual(triggerBox.left);
-        expect(dropdownBox.width).toBeGreaterThanOrEqual(triggerBox.width);
-      })();
+      test.each(alignments)('alignment at %s corner', alignment => {
+        return setupTest(async page => {
+          const wrapper = createWrapper().findAutosuggest(`#${alignment}`);
+          await page.click(wrapper.findNativeInput().toSelector());
+          const dropdownBox = await page.getBoundingBox(wrapper.findDropdown({ expandToViewport }).toSelector());
+          const triggerBox = await page.getBoundingBox(wrapper.findNativeInput().toSelector());
+          if (alignment.endsWith('-left')) {
+            expect(dropdownBox.left).toEqual(triggerBox.left);
+          } else {
+            expect(dropdownBox.right).toEqual(triggerBox.right);
+          }
```

Comment:
Fixing the test, because it was testing the wrong behavior.

On the left side alignment, we want to test this:

<img width="310" alt="image" src="https://github.com/user-attachments/assets/e490a0b4-1ea8-4828-91f1-acf5f8334100">

On the right side alignment, we want to test this

<img width="293" alt="image" src="https://github.com/user-attachments/assets/b65cc803-dafb-4012-a000-26e127b1942f">

But instead, we were testing left alignment twice, which is useless. Fixed the dev page and the test to fix the correct use-case